pkg/controller/externaldata/stats_reporter.go

```diff
@@ -0,0 +1,116 @@
+package externaldata
+
+import (
+	"context"
+	"sync"
+
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/metrics"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	providerMetricName     = "providers"
+	providerErrorCountName = "provider_error_count"
+	statusKey              = "status"
+
+	providerDesc      = "Number of external data providers by status"
+	providerErrorDesc = "Incremental counter for all provider errors occurring over time"
+)
+
+var providerErrorCountM metric.Int64Counter
+
+func (r *reporter) observeProviderMetric(_ context.Context, o metric.Int64Observer) error {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	if !r.dirty {
+		return nil
+	}
+	for status, count := range r.statusReport {
+		o.Observe(count, metric.WithAttributes(attribute.String(statusKey, string(status))))
+	}
+	r.dirty = false
+	return nil
+}
+
+// newStatsReporter creates a reporter for external data provider metrics.
+func newStatsReporter() *reporter {
+	var err error
+	r := &reporter{
+		cache: make(map[types.NamespacedName]metrics.Status),
+	}
+	meter := otel.GetMeterProvider().Meter("gatekeeper")
+
+	// Register the gatekeeper_providers gauge metric
+	_, err = meter.Int64ObservableGauge(
+		providerMetricName,
+		metric.WithDescription(providerDesc),
+		metric.WithInt64Callback(r.observeProviderMetric),
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	// Register the gatekeeper_provider_error_count counter metric
+	providerErrorCountM, err = meter.Int64Counter(
```

Comment:
can you make sure to add these to the docs too

Reply:
updated docs and added this info.